src/api_server.cc

```diff
@@ -56,7 +61,11 @@ void MetadataApiServer::Handler::operator()(const HttpServer::request& request,
       if (config_.VerboseLogging()) {
         LOG(WARNING) << "No matching resource for " << id;
       }
+      std::map<std::string, std::string> headers = {
+        {"Content-Type", "text/plain"},
```

Comment:
I've chosen to use text/plain here as we're not actually returning JSON. If we decide we'd like to return an empty object, we can update this accordingly.

Reply:
Sounds good. Should we also write `"Not found\n"` into the response?

Reply:
Yeah. I've found some examples that also add `"code":404`. I'm happy with either one -- we are, in effect, designing this API. Let's do it!